docs/content/docs/dev/configuration/advanced.md

```diff
@@ -0,0 +1,158 @@
+In IntelliJ IDEA, the recommended way to change JVM options is from the `Help | Edit Custom VM Options` menu.
+See [this article](https://intellij-support.jetbrains.com/hc/en-us/articles/206544869-Configuring-JVM-options-and-platform-properties) for details.
+
+**Note on IntelliJ:** To make the applications run within IntelliJ IDEA, it is necessary to tick the
+`Include dependencies with "Provided" scope` box in the run configuration. If this option is not available
+(possibly due to using an older IntelliJ IDEA version), then a workaround is to create a test that
+calls the application's `main()` method.
+
+## Scala Versions
+
+Different Scala versions are not binary compatible with one another. All Flink dependencies that 
+(transitively) depend on Scala are suffixed with the Scala version that they are built for 
+(i.e. `flink-streaming-scala_2.12`).
+
+If you are only using Flink's Java APIs, you can use any Scala version. If you are using Flink's Scala APIs, 
+you need to pick the Scala version that matches the application's Scala version.
+
+Please refer to the [build guide]({{< ref "docs/flinkDev/building" >}}#scala-versions) for details 
+on how to build Flink for a specific Scala version.
+
+Scala versions after 2.12.8 are not binary compatible with previous 2.12.x versions. This prevents
+the Flink project from upgrading its 2.12.x builds beyond 2.12.8. You can build Flink locally for
+later Scala versions by following the [build guide]({{< ref "docs/flinkDev/building" >}}#scala-versions).
+For this to work, you will need to add `-Djapicmp.skip` to skip binary compatibility checks when building.
+
+See the [Scala 2.12.8 release notes](https://github.com/scala/scala/releases/tag/v2.12.8) for more details.
+The relevant section states:
+
+> The second fix is not binary compatible: the 2.12.8 compiler omits certain
+> methods that are generated by earlier 2.12 compilers. However, we believe
+> that these methods are never used and existing compiled code will continue to
+> work.  See the [pull request
+> description](https://github.com/scala/scala/pull/7469) for more details.
+
+
+The Flink distribution contains by default the required JARs to execute Flink SQL Jobs (found in the `/lib` folder), 
+in particular:
+
+-`flink-table-api-java-uber-{{< version >}}.jar` --> contains all the Java APIs
+-`flink-table-runtime-{{< version >}}.jar` --> contains the runtime
+-`flink-table-planner-loader-{{< version >}}.jar` --> contains the query planner
+
+**Note:** Previously, these JARs were all packaged into `flink-table.jar`. This has now been split into 
+three JARs in order to allow users to swap the `flink-table-planner-loader-{{< version >}}.jar` with
+`flink-table-planner{{< scala_version >}}-{{< version >}}.jar`.
+
+When using formats and connectors with the Flink Scala API, you need to either download and manually 
+include these JARs in the `/lib` folder (recommended), or you need to shade them in the uber JAR of your 
+Flink SQL Jobs.
```

Comment:
This sentence is incorrect, I think i badly worded it initially. What I was trying to explain is that while for java apis, everything is already in the distribution so the dependency can be "provided" (as explained in the overview and maven page), for scala apis you don't have the jar built-in in the distribution, so you need to package the jar somehoe, effectively like a connector dependency.

Perhaps reword this to something like:

> While Table Java APIs artifacts are built-in in the distribution, Table Scala API artifacts are not included by default. You need to either include them in the distribution `/lib` folder manually (recommended), or package them as dependencies in your job uber jar.